core/forkid/forkid.go

```diff
@@ -297,5 +292,9 @@ func gatherForks(config *params.ChainConfig) ([]uint64, []uint64) {
 	if len(forksByTime) > 0 && forksByTime[0] == 0 {
 		forksByTime = forksByTime[1:]
 	}
+	// Skip any forks by time that are non-zero, but before genesis.
```

Comment:
Shoulnd't we drop the 0 time check above? The new code already encompasses the old one

Reply:
you are correct